Resolves "Panzer:  Update Integrator_GradBasisCrossVector"

jmgate opened the pull request:
@trilinos/panzer 

## Description
Switch `Integrator_GradBasisCrossVector` to support `CONTRIBUTES`-style evaluations, and Kokkos-ify.

## Motivation and Context
Closes #1892 

## How Has This Been Tested?
I've tested Panzer locally.  We'll see how the @trilinos-autotester does.

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

## DO NOT MERGE

Once this PR has been reviewed and approved, I will squash before merging.

jmgate commented:
@rppawlo, another major class refactor.  The substantive commits to look at are 163d984 (create main and delegating constructor) and e2424b6 (create operator()()).

jmgate commented:
Just realized the missing `PHX::Device::fence()`.  Adding that now&hellip;

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 1366</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c9c5918281291d8e98473f50ae0ce7c1d6efac77</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a86e4827888aab4c87ae9f1f28b65e7894f309b3</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1929</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c9c5918281291d8e98473f50ae0ce7c1d6efac77</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a86e4827888aab4c87ae9f1f28b65e7894f309b3</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 1591</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c9c5918281291d8e98473f50ae0ce7c1d6efac77</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a86e4827888aab4c87ae9f1f28b65e7894f309b3</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (jmgate/Trilinos)</dt>
  <ul>
    <li>Branch: 1892-panzer-update-integrator-grad-basis-cross-vector</li>
    <li>SHA: c9c5918281291d8e98473f50ae0ce7c1d6efac77</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: jmgate
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 1366</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c9c5918281291d8e98473f50ae0ce7c1d6efac77</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a86e4827888aab4c87ae9f1f28b65e7894f309b3</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1929</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c9c5918281291d8e98473f50ae0ce7c1d6efac77</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a86e4827888aab4c87ae9f1f28b65e7894f309b3</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 1591</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>c9c5918281291d8e98473f50ae0ce7c1d6efac77</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>a86e4827888aab4c87ae9f1f28b65e7894f309b3</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 3616.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-3616-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO REVIEWS HAVE BEEN PERFORMED ON THIS PULL REQUEST!

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

trilinos-autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

jmgate commented:
@rppawlo, squashed and rebased.  Can you re-approve?  I wonder if there's a smart way to modify the @trilinos-autotester such that we don't need to do this double approval business.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 1372</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>278aaeefa62346623cf31552f0f10331800739b7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>72985ec68786b9f35758d4482ce5aa5957dc7ae2</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1935</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>278aaeefa62346623cf31552f0f10331800739b7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>72985ec68786b9f35758d4482ce5aa5957dc7ae2</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 1597</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>278aaeefa62346623cf31552f0f10331800739b7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>72985ec68786b9f35758d4482ce5aa5957dc7ae2</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (jmgate/Trilinos)</dt>
  <ul>
    <li>Branch: 1892-panzer-update-integrator-grad-basis-cross-vector</li>
    <li>SHA: 278aaeefa62346623cf31552f0f10331800739b7</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: jmgate
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_intel_17.0.1</h3>
<ul>
  <li>Build Num: 1372</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>278aaeefa62346623cf31552f0f10331800739b7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>72985ec68786b9f35758d4482ce5aa5957dc7ae2</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.9.3</h3>
<ul>
  <li>Build Num: 1935</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.9.3</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>278aaeefa62346623cf31552f0f10331800739b7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>72985ec68786b9f35758d4482ce5aa5957dc7ae2</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_pullrequest_gcc_4.8.4</h3>
<ul>
  <li>Build Num: 1597</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>COMPILER_MODULE</td>
    <td>sems-gcc/4.8.4</td>
  </tr>

  <tr>
    <td>JENKINS_BUILD_TYPE</td>
    <td>Release</td>
  </tr>

  <tr>
    <td>JENKINS_COMM_TYPE</td>
    <td>MPI</td>
  </tr>

  <tr>
    <td>JENKINS_DO_COMPLEX</td>
    <td>OFF</td>
  </tr>

  <tr>
    <td>JENKINS_JOB_TYPE</td>
    <td>Experimental</td>
  </tr>

  <tr>
    <td>MPI_MODULE</td>
    <td>sems-openmpi/1.8.7</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>3616</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>1892-panzer-update-integrator-grad-basis-cross-vector</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/jmgate/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>278aaeefa62346623cf31552f0f10331800739b7</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>72985ec68786b9f35758d4482ce5aa5957dc7ae2</td>
  </tr>
</table>
</details>


<br>[CDash Test Results for PR# 3616.](https://testing-vm.sandia.gov/cdash/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-3616-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

trilinos-autotester commented:
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ rppawlo ]!

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

trilinos-autotester commented:
Merge on Pull Request# 3616: IS A SUCCESS - Pull Request successfully merged
